graphene/src/document.rs

```diff
@@ -375,6 +375,23 @@ impl Document {
 		Ok(())
 	}
 
+	fn mark_layer_type_as_dirty(root: &mut Layer, data_type: LayerDataTypeDiscriminant) -> bool {
+		if LayerDataTypeDiscriminant::from(&root.data) == data_type {
+			root.cache_dirty = true;
+		} else if let LayerDataType::Folder(f) = &mut root.data {
+			let mut dirty = false;
+			for l in f.layers_mut() {
```

Comment:
Slight preference for full words instead of single-letter variable names, although not a big deal in small-coverage areas like this. Feel free to disregard this if you disagree regarding cleanliness of the code.